### ISSUE TITLE
Feature request: Support proxy mode

### ISSUE THREAD

**Issue by NobodyXu** [NobodyXu/openssh-mux-client#6]
This would enable this crate to be used on windows 10, where unix domain socket is supproted but sending fds over it is not supported.

This would also make it possible to pool connection and further improve performance, at the cost of more complex code and more code to maintain.

But I think the benefit outweight the cost.

**Comment by NobodyXu** [comment]
The windows support is blocked on https://github.com/tokio-rs/tokio/issues/2201

**Comment by NobodyXu** [comment]
Superseded by openssh-rust/openssh#152 which avoids external ssh dependency altogether
